### PR TITLE
[SNAP-2582] allow for NONE as a valid policy for server-auth-provider

### PR DESCRIPTION
**Pull request by sumwale**
These set of changes (along with those in store-side PR) allow one to specify server-auth-provider=NONE so that gemfire-level p2p authentication is disabled while the auth-provider will set the connection-level authentication (both for client connections and internal connections).

## Changes proposed in this pull request

- allow for NONE as a valid security policy in Lead boot
- fixed and cleaned up lead property handling to set the properties correctly for start and stop
- updated calls from snappy layer to use the authentication service from the Database module
  rather than the static version (which has not been removed on the store side)
- wait for sampler to fully initialiaze before proceeding with hive catalog initialization
- changed tests extending ClusterManagerLDAPTestBase to use server-auth-provider=NONE
  and test that authentication has been disabled at jgroups AUTH layer

## Patch testing

precheckin with the new tests; manual testing

## ReleaseNotes.txt changes

Document the possibility of setting "-server-auth-provider=NONE" to disable peer-to-peer authentication

## Other PRs 

https://github.com/SnappyDataInc/snappy-store/pull/429

### CODE REVIEW

File: core/src/main/java/io/snappydata/impl/SnappyHiveCatalog.java [hunk 1]
```diff
@@ -633,6 +633,8 @@ private void initHMC() {
           Constant.JDBC_EMBEDDED_DRIVER());
       initCommonHiveMetaStoreProperties(metadataConf);
 
+      // wait for stats sampler initialization
+      Misc.waitForSamplerInitialization();
```
Comment (kneeraj) [inline core/src/main/java/io/snappydata/impl/SnappyHiveCatalog.java]: Just curious. Why is this required?

Comment (sumwale) [reply core/src/main/java/io/snappydata/impl/SnappyHiveCatalog.java]: As mentioned in other PR comments, saw some stats failures in region initialization so this ensures that stats sampler is up before opening up the catalog for users.
